File: relisoft-hr/server/models/EmployeeTeam.js
import mongoose from 'mongoose';

const employeeTeamSchema = new mongoose.Schema({
  employee: { type: mongoose.Schema.Types.ObjectId, ref: 'Employee', required: true },
  team: { type: mongoose.Schema.Types.ObjectId, ref: 'Team', required: true },
}, { timestamps: true });

employeeTeamSchema.index({ employee: 1, team: 1 }, { unique: true });

export default mongoose.model('EmployeeTeam', employeeTeamSchema);

File: relisoft-hr/server/Models/EmployeeTeam.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RelisoftHR.Models;

public class EmployeeTeam
{
    [Key]
    public int Id { get; set; }
    public int EmployeeId { get; set; }
    public int TeamId { get; set; }

    [ForeignKey(nameof(EmployeeId))]
    public Employee? Employee { get; set; }
    [ForeignKey(nameof(TeamId))]
    public Team? Team { get; set; }
}
